packages/live-share-inking/src/core/InkingManager.ts

```diff
@@ -0,0 +1,1293 @@
+ */
+export interface IWetStroke extends IStroke {
+    /**
+     * The type of the wet stroke.
+     */
+    readonly type: StrokeType;
+    /**
+     * Ends the wet stroke.
+     * @param p Optional. The points at which the stroke ends. If not specified,
+     * the stroke ends at the last added point.
+     */
+    end(): void;
+    /**
+     * Cancels the wet stroke.
+     */
+    cancel(): void;
+}
+
+class ChangeLog {
+    private _addedStrokes: Map<string, IStroke> = new Map<string, IStroke>();
+    private _removedStrokes: Set<string> = new Set<string>();
+
+    public clear() {
+        this._addedStrokes.clear();
+        this._removedStrokes.clear();
+    }
+
+    public mergeChanges(changes: ChangeLog) {
+        for (let id of changes._removedStrokes) {
+            if (!this._addedStrokes.delete(id)) {
+                this._removedStrokes.add(id);
+            }
+        }
+
+        changes._addedStrokes.forEach(
+            (value: IStroke) => {
+                this._addedStrokes.set(value.id, value);
+            });
+    }
+
+    public addStroke(stroke: IStroke) {
+        this._addedStrokes.set(stroke.id, stroke);
+    }
+
+    public removeStroke(id: string) {
+        this._removedStrokes.add(id);
+    }
+
+    public getRemovedStrokes(): string[] {
+        return [...this._removedStrokes];
+    }
+
+    public getAddedStrokes(): IStroke[] {
+        return [...this._addedStrokes.values()];
+    }
+
+    get hasChanges(): boolean {
+        return this._addedStrokes.size > 0 || this._removedStrokes.size > 0;
+    }
+}
+
+class EphemeralCanvas extends DryCanvas {
+    private _removalTimeout?: number;
+
+    constructor(readonly clientId: string, parentElement?: HTMLElement) {
+        super(parentElement)
+    }
+
+    scheduleRemoval(onRemoveCallback: (canvas: EphemeralCanvas) => void) {
+        if (this._removalTimeout) {
+            window.clearTimeout(this._removalTimeout);
+        }
+
+        this._removalTimeout = window.setTimeout(
+            () => {
+                this.fadeOut();
+
+                onRemoveCallback(this);
+            },
+            InkingManager.ephemeralCanvasRemovalDelay
+        )
+
+    }
+}
+
+/**
+ * Defines options used by `InkingManager.addStroke` and `InkingManager.removeStroke`.
+ */
+export interface IAddRemoveStrokeOptions {
+    /**
+     * Optional. Indicates if the canvas must be fully re-rendered at once after the
+     * stroke has been added or removed. Defaults to `false`.
+     */
+    forceReRender?: boolean,
+    /**
+     * Optional. Indicates whether the add or remove operation should be added to the
+     * change log, which in turn will lead to `StrokeAddedEvent` or `StrokeRemovedEvent`
+     * begin emitted. Defaults to `true`.
+     */
+    addToChangeLog?: boolean
+}
+
+/**
+ * Handles user interaction with a canvas, and manages the rendering of wet and dry strokes.
+ */
+export class InkingManager extends EventEmitter {
```

Comment:
>EventEmitter [](http://example.com/codeflow?start=35&length=12)

You might want to consider deriving from fluids `TypedEventEmitter` class as devs will get intelisense and type checking for all event subscriptions.  Look at `EphemeralEventScope` for an example.

Reply:
@Stevenic  I can't do that, I want the inking stuff to remain standalone. `InkingManager` doesn't require any of the Fluid/Live Share stuff. Only `SharedInkingSession` does. That makes it possible to split the two extremely easily (and I hope we will).